pulsar-broker/src/main/java/org/apache/pulsar/broker/service/ServerCnx.java

```diff
@@ -543,10 +545,7 @@ protected void handleConnect(CommandConnect connect) {
                 connect.hasOriginalPrincipal() ? connect.getOriginalPrincipal() : null,
                 sslSession);
 
-            AuthenticationDataSource authenticationData = authenticationProvider
-                .newAuthDataSource(clientData, remoteAddress, sslSession);
-            authState = authenticationProvider.newAuthState(authenticationData);
-
+            authState = authenticationProvider.newAuthState(clientData, remoteAddress, sslSession);
```

Comment:
clientData is being used twice here. It would make more sense to remove it from the call to newAuthState and only use it via the call to doAuthentication.

Reply:
Thanks. newAuthState need clientData, it is a kind of init parameter for AuthState, especially in OneStageAuth.